prow/config/config.go

```diff
@@ -831,6 +837,27 @@ func validateJobBase(v JobBase, jobType prowapi.ProwJobType, podNamespace string
 	return validateDecoration(v.Spec.Containers[0], v.DecorationConfig)
 }
 
+func (c *Config) validatePresubmits(repo string, additionalPresubmits ...Presubmit) error {
+	var validPresubmits []Presubmit
+	for _, presubmit := range append(c.presubmits[repo], additionalPresubmits...) {
+		if err := validateJobBase(presubmit.JobBase, prowapi.PresubmitJob, c.PodNamespace); err != nil {
+			return fmt.Errorf("invalid presubmit job %s: %v", presubmit.Name, err)
+		}
+
+		if err := validateTriggering(presubmit); err != nil {
+			return err
+		}
+
+		for _, validatedPresubmit := range validPresubmits {
```

Comment:
This needs to be qualified by job name otherwise we wouldn't allow more than one job on the any branch.
We also need to actually append things to the list of valid presubmits for the job name. Right now it is always empty.